presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -82,17 +80,28 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
     }
 
     @Override
-    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorTableHandle handle, Constraint<ColumnHandle> constraint)
     {
-        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new JdbcTableLayoutHandle(tableHandle, constraint.getSummary()));
-        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+        JdbcTableHandle jdbcHandle = (JdbcTableHandle) handle;
+        jdbcHandle = new JdbcTableHandle(
+                jdbcHandle.getSchemaTableName(),
+                jdbcHandle.getCatalogName(),
+                jdbcHandle.getSchemaName(),
+                jdbcHandle.getTableName(),
+                constraint.getSummary());
+        return Optional.of(new ConstraintApplicationResult<>(jdbcHandle, constraint.getSummary()));
```

Comment:
Why is the remainingFilter set to `constraint.getSummary()`? Doesn't this connector guarantee the rows will match that criteria?

Reply:
No, there are issues with case sensitivity for MySQL and maybe others. It’s also trading safety for performance, which shouldn’t matter given the overhead of JDBC.

Reply:
It should be possible to test. You can define a MySQL table with case-insensitive collation.
It's not the first time that the idea to slim down the `remainingFilter ` occurs.

Reply:
I agree it's possible, but at this point I don't trust the filtering code to be 100% for all databases, and doing so is outside the scope of this PR.

Reply:
Also note that this has the same behavior has the previous code (return original constraint).